File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/WorkOrder.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// WorkOrder Declaration
//==============================================================
type WorkOrder struct {
    gorm.Model
     WorkOrderNumber                                    string
    PlannedStart                                                            time.Time
    PlannedEnd                                                            time.Time
    Quantity                                                            string
    Priority                                                            string
    ItemId         *uint
    Item           *Item `gorm:"foreignKey:ItemId"`
    PlantId         *uint
    Plant           *Plant `gorm:"foreignKey:PlantId"`
    RoutingId         *uint
    Routing           *Routing `gorm:"foreignKey:RoutingId"`
    BomId         *uint
    Bom           *BOM `gorm:"foreignKey:BomId"`
    ProductionScheduleId         *uint
    ProductionSchedule           *ProductionSchedule `gorm:"foreignKey:ProductionScheduleId"`
    SalesOrderId         *uint
    SalesOrder           *SalesOrder `gorm:"foreignKey:SalesOrderId"`
    Status                      WorkOrderStatus

// parent associations as their child

}

File: evidence/generated-systems/angular/manufacturing-on-angular/src/app/models/WorkOrder.js

// Define collection and schema for WorkOrder
export interface WorkOrder {
    workOrderNumber:
	type : string
    plannedStart:
	type : Date
    plannedEnd:
	type : Date
    quantity:
	type : Quantity
    priority:
	type : number
    Item:
	type : Schema.Types.ObjectId
    Plant:
	type : Schema.Types.ObjectId
    Routing:
	type : Schema.Types.ObjectId
    Bom:
	type : Schema.Types.ObjectId
    ProductionSchedule:
	type : Schema.Types.ObjectId
    SalesOrder:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'workOrders'
}
